antmonitorlib/src/main/java/edu/uci/calit2/antmonitor/lib/vpn/VpnClient.java

```diff
@@ -366,7 +407,18 @@ public ParcelFileDescriptor buildTunInterface() {
         builder.addRoute("0.0.0.0", 0);
 
         // Use Google's DNS server
-        builder.addDnsServer("8.8.8.8");
+        builder.addDnsServer(DNS_SERVER);
+
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+            for (String excludedApp: excludedApps) {
+                try {
+                    builder.addDisallowedApplication(excludedApp);
+                } catch (PackageManager.NameNotFoundException ex) {
+                    //
```

Comment:
Consider printing a warning here (perhaps at a debug level), to follow the rest of the AntMonitor style.